pkg/apis/cr/v1alpha1/types.go

```diff
@@ -225,77 +254,112 @@ type BlueprintPhase struct {
 type BlueprintList struct {
 	metav1.TypeMeta `json:",inline"`
 	metav1.ListMeta `json:"metadata"`
-	Items           []*Blueprint `json:"items"`
+	// Items is the list of Blueprints.
+	Items []*Blueprint `json:"items"`
 }
 
 // +genclient
 // +genclient:noStatus
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-// Profile
+// Profile captures information about a location for data operation artifacts and
+// corresponding credentials that will be made available to a Blueprint.
 type Profile struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata"`
-	Location          Location   `json:"location"`
-	Credential        Credential `json:"credential"`
-	SkipSSLVerify     bool       `json:"skipSSLVerify"`
+	// Specifies the location that the Blueprint can use.
+	Location Location `json:"location"`
+	// Represents the credentials associated with the Location.
+	Credential Credential `json:"credential"`
+	// SkipSSLVerify is boolean and specifies whether skipping SkipSSLVerify verification
+	// is allowed when operating with the Location.
+	// If omitted from a CR definition it default to false
+	SkipSSLVerify bool `json:"skipSSLVerify"`
 }
 
 // LocationType
 type LocationType string
 
 const (
-	LocationTypeGCS         LocationType = "gcs"
+	// LocationType is selected as GCS.
+	LocationTypeGCS LocationType = "gcs"
+	// LocationType is selected as S3 Complaint.
 	LocationTypeS3Compliant LocationType = "s3Compliant"
-	LocationTypeAzure       LocationType = "azure"
-	LocationTypeKopia       LocationType = "kopia"
+	// LocationType is selected as Azure.
+	LocationTypeAzure LocationType = "azure"
+	// LocationType is selected as Kopia.
+	LocationTypeKopia LocationType = "kopia"
 )
 
 // Location
 type Location struct {
-	Type     LocationType `json:"type"`
-	Bucket   string       `json:"bucket"`
-	Endpoint string       `json:"endpoint"`
-	Prefix   string       `json:"prefix"`
-	Region   string       `json:"region"`
+	// Type of the Location being used.
+	Type LocationType `json:"type"`
```

Comment:
Type specifies, the kind of object storage that would be used to upload the backup objects.

Reply:
Let's also specify what are all the types of locations supported. maybe refer to them somehow.

Reply:
How does this comment sound like :
Type specifies the kind of object storage that would be used to upload the backup objects. For example: "GCS" or "S3Compliant" or "Azure" or "Kopia" .

Reply:
instead of for example I think we should say, currently supported values are.

Reply:
Alright noted.